### DISCUSSION TITLE
confused by logic of "FundMe"

### DISCUSSION THREAD

**Discussion by frank0532** [smartcontractkit/full-blockchain-solidity-course-py#1670] [Q&A] [answered]
At first I select 'account_1' to deploy this contract, and then "fund" 1e+18 Wei  by 'account_1'; after that I switch to "account_2" and "fund" 1e+18 Wei again.
After these two transactions, "addressToAmountFunded" and "funders" both records these accounts' funding, but the "owner" is account_1. Then I have some questions as below:
1. after these two transactions, which account receives these two funding?
2. when we switch to "account_1", and execute "withdraw", then account_1 can receive its fuding but account_2 can't , and variables( "addressToAmountFunded" and "funders") are reseted to initiate status, but I think  variables should hold value of account_2, because account_2 not withdraw.

**Comment by cromewar** [comment, marked as answer]
Hello @frank0532 
1. The funding goes to the address of the contract `Fundme` which keeps the amount funded, so neither the account 1 or 2 have that funds.
2. The withdraw logic is to retrieve all the funds stored on the contract `Fundme` so if 10 accounts send funds, the one using the withdraw function, will get all the funds (of the 10 accounts). 
Here you have to take something in consideration, we suppose the creator of the crowdfunding campaign is the same one who deploys the contract, so all the funds are withdraw to the account_1 (as is the owner of the contract).
But let's suppose the person creating the contract is not the same as the one requesting the funding, well there are a lot of solutions here (quite more advanced to explain), but the easiest way is to add and address to the withdraw function to specify which account is going to get all the funds.

I hope this info might help.

**Reply by frank0532** [reply]
Hello @cromewar 
Thanks a lot for your explanation by which I have cleared much about this.
Maybe I should consider the address of the contract as an account which is very similar with account_1/2, yes?

**Reply by cromewar** [reply]
well depends on how your protocol works, in this case as the account one is the same which deploys de contract, so yes.
